client.go

```diff
@@ -581,3 +585,80 @@ func nonce() []byte {
 	}
 	return nonce
 }
+
+func (c *Client) GetOidcPollCode(ctx context.Context, logger logrus.FieldLogger) (string, error) {
+	logger.WithFields(logrus.Fields{"server": c.dnServer}).Debug("Making GetOidcPollCode request to API")
+
+	enrollURL, err := url.JoinPath(c.dnServer, message.PreAuthEndpoint)
+	if err != nil {
+		return "", err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", enrollURL, nil)
+	if err != nil {
+		return "", err
+	}
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	// Log the request ID returned from the server
+	reqID := resp.Header.Get("X-Request-ID")
+	l := logger.WithFields(logrus.Fields{"statusCode": resp.StatusCode, "reqID": reqID})
+	b, err := c.handleBody(resp)
+	if err != nil {
+		l.Error(err) //todo I don't like erroring and also logging?
+		return "", err
+	}
+
+	// Decode the response
+	r := message.PreAuthResponse{}
+	if err = json.Unmarshal(b, &r); err != nil {
+		return "", &APIError{e: fmt.Errorf("error decoding JSON response: %s\nbody: %s", err, b), ReqID: reqID}
+	}
+
+	return r.PollToken, nil
+}
+
+func (c *Client) DoOidcPoll(ctx context.Context, logger logrus.FieldLogger, pollCode string) (*message.EnduserAuthPollResponse, error) {
+	logger.WithFields(logrus.Fields{"server": c.dnServer}).Debug("Making DoOidcPoll request to API")
+
+	enrollURL, err := url.JoinPath(c.dnServer, message.EnduserAuthPoll)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "GET", enrollURL, nil)
+	if err != nil {
+		return nil, err
+	}
+	q := req.URL.Query()
+	q.Add("token", pollCode)
+	req.URL.RawQuery = q.Encode()
```

Comment:
nit: It feels a little goofy to modify the `req.URL.RawQuery` of the request instead of just setting the query string on `enrollURL` before passing it. :-)

```
enrollURL = enrollURL + "?" + url.Values{"token": pollCode}.Encode()
```

Or build up a URL object instead of using JoinPath & url.Values directly. :-)